flipt-csharp/src/Clients/Evaluation.cs

```diff
@@ -0,0 +1,155 @@
+                return this;
+            }
+
+            public Evaluation Build()
+            {
+                if (HttpClient == null)
+                {
+                    throw new InvalidOperationException("HttpClient must be provided.");
+                }
+
+                if (string.IsNullOrEmpty(BaseUrl))
+                {
+                    throw new InvalidOperationException("BaseURL must be provided.");
+                }
+
+                return new Evaluation(this);
+            }
+        }
+
+        /// <summary>
+        /// This method evaluates a variant for a given flag key and entity id.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public async Task<VariantEvaluationResponse?> EvaluateVariantAsync(EvaluationRequest request)
+        {
+            return await EvaluateAsync<VariantEvaluationResponse>("/evaluate/v1/variant", request);
+        }
+
+        /// <summary>
+        /// This method evaluates a boolean flag for a given flag key and entity id.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public async Task<BooleanEvaluationResponse?> EvaluateBooleanAsync(EvaluationRequest request)
+        {
+            return await EvaluateAsync<BooleanEvaluationResponse>("/evaluate/v1/boolean", request);
+        }
+
+        /// <summary>
+        /// This method evaluates a batch of flags for a given flag key and entity id.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public async Task<BatchEvaluationResponse?> EvaluateBatchAsync(BatchEvaluationRequest request)
+        {
+            return await EvaluateAsync<BatchEvaluationResponse>("/evaluate/v1/batch", request);
+        }
+
+        /// <summary>
+        /// This method evaluates a flag for a given flag key and entity id.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="path"></param>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        private async Task<T?> EvaluateAsync<T>(string path, object request)
+        {
+            var url = new Uri(new Uri(_baseUrl), path);
+
+            var jsonContent = JsonSerializer.Serialize(request);
+            var content = new StringContent(jsonContent, Encoding.UTF8, "application/json");
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, url)
+            {
+                Content = content
+            };
+
+            if (_headers != null)
+                foreach (var header in _headers)
+                {
+                    httpRequest.Headers.Add(header.Key, header.Value);
+                }
+
+            if (_authenticationStrategy != null)
+                httpRequest.Headers.Authorization =
+                    new AuthenticationHeaderValue("Bearer", _authenticationStrategy.GetAuthorizationHeader());
```

Comment:
I dont think we want `Bearer` here always because depending on the authorization strategy, the header should be:

`Authorization: Bearer ${token}` for client token strategy : https://docs.flipt.io/authentication/using-tokens

or 

`Authorizatoin: JWT ${token}` for JWT strategy: https://docs.flipt.io/authentication/using-jwts

I know the JWT prefix isn't standard, but we wanted our server to easily know which type of token it was for decoding the auth instead of first trying static token then JWT as we would have to do if we used the `Bearer` prefix for both

Reply:
fixed